### PR TITLE
feat: add timeline filter

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
This PR delete MIgration to workflow views

1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: engine/elasticsearch/elasticsearch_handlers.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ import (
 
 	"gopkg.in/olivere/elastic.v5"
 
+	"fmt"
```
Comment (golangci) [inline engine/elasticsearch/elasticsearch_handlers.go]: File is not goimports-ed

File: sdk/event.go [hunk 2]
```diff
@@ -35,7 +35,8 @@ type Event struct {
 
 // EventFilter represents filters when getting events
 type EventFilter struct {
-	ProjectKeys []string `json:"project_keys"`
+	CurrentItem int            `json:"currentItem"`
```
Comment (bnjjj) [inline sdk/event.go]: why not json in snake case as the others ?

File: ui/src/app/views/home/timeline/home.timeline.component.ts [hunk 3]
```diff
@@ -16,19 +18,47 @@ export class HomeTimelineComponent implements OnInit {
...
+    selectTab(t: string): void {
+        this.selectedTab = t;
+    }
+
+    onScroll() {
+        this._timelineStore.getMore(this.currentItem + 1, false);
+    }
+
+    updateFilter(): void {
+        console.log('BIM');
```
Comment (bnjjj) [inline ui/src/app/views/home/timeline/home.timeline.component.ts]: BAM BOOM

File: ui/src/app/views/project/list/project.list.component.ts [hunk 4]
```diff
@@ -20,12 +19,15 @@ export class ProjectListComponent {
         });
     }
 
-    constructor(private _projectService: ProjectService) {
-        this._projectService.getProjects(false)
-            .pipe(finalize(() => this.loading = false))
+    constructor(private _projectStore: ProjectStore) {
+        // TODO NEVER CALL ProjectSERVICE
```
Comment (bnjjj) [inline ui/src/app/views/project/list/project.list.component.ts]: delete TODO

File: ui/src/app/views/project/list/project.list.component.ts [hunk 5]
```diff
@@ -20,12 +19,15 @@ export class ProjectListComponent {
...
+        this._projectStore.getProjects()
             .subscribe((projects) => {
-                this.projects = projects;
-                this.filteredProjects = projects;
+                if (projects) {
+                    this.loading = false;
```
Comment (bnjjj) [inline ui/src/app/views/project/list/project.list.component.ts]: loading in finalize ?

File: ui/src/assets/i18n/fr.json [hunk 6]
```diff
@@ -642,6 +642,13 @@
   "timeline_loading": "Chargement du fil d'actualité...",
   "timeline_no": "Le fil d'actualité est vide",
   "timeline_title": "Fil d'actualité",
+  "timeline_filter_all_project": "Tous les projets",
+  "timeline_filter_project": "Recevoir des evenements sur :",
```
Comment (bnjjj) [inline ui/src/assets/i18n/fr.json]: s/evenements/évènements

File: ui/src/app/views/project/list/project.list.component.ts (hunk 5 above)
Comment (sguiheux) [reply ui/src/app/views/project/list/project.list.component.ts]: There is no end

File: ui/src/assets/i18n/fr.json (hunk 6 above)
Comment (antleblanc) [reply ui/src/assets/i18n/fr.json]: s/évènements/événements :)

File: engine/api/timeline.go [hunk 7]
```diff
@@ -5,19 +5,39 @@ import (
...
 	"github.com/ovh/cds/sdk"
 )
 
 func (api *API) getTimelineHandler() Handler {
 	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+
+		currentItemString := FormString(r, "currentItem")
 		u := getUser(ctx)
+		if currentItemString == "" {
+			currentItemString = "0"
```
Comment (golangci) [inline engine/api/timeline.go]: ineffectual assignment to `currentItemString`

**Pull request merged**